pkgs/build-support/build-graalvm-native-image/default.nix

```
@@ -49,6 +49,11 @@ stdenv.mkDerivation ({
 
   nativeImageBuildArgs = nativeImageBuildArgs ++ extraNativeImageBuildArgs ++ [ graalvmXmx ];
 
+  # Workaround GraalVM issue where the builder does not have access to the
+  # environment variables since 21.0.0
+  # https://github.com/oracle/graal/pull/6095
```

```suggestion
  # https://github.com/oracle/graal/pull/6095
  # https://github.com/oracle/graal/issues/7502
```
I'd add a link to the issue you created, I think the information there is useful